flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryImpl.java

```diff
@@ -90,7 +92,9 @@
 	/**
 	 * Creates a new MetricRegistry and starts the configured reporter.
 	 */
-	public MetricRegistryImpl(MetricRegistryConfiguration config) {
+	public MetricRegistryImpl(Configuration configuration) {
+		MetricRegistryConfiguration config = MetricRegistryConfiguration.fromConfiguration(configuration);
```

Comment:
the whole point of the `MetricRegistryConfiguration` is to not expose the `Configuration` directly to the `MetricRegistry`. This should be reverted. Instead, extend the MRConfiguration with a separate configuration object for the akka stuff that the `MetricQueryService` requires, which btw. can be determined before the actor is even started.